### ISSUE TITLE
Two training sessions, increased losses

### ISSUE THREAD

**Issue by sonyoungrack** [ultralytics/ultralytics#7786]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am a Korean developer. I can speak a little English, but please understand that I am writing it using a translator.

I trained the yolov8n model 200 times in colab environment. My dataset is big, so it takes a very long time, so I checked that cls_loss goes down to about 0.3, and after I saved the model, I tried to train it 200 more times the next day. But even though the dataset is the same and the last model is loaded on the model, cls_loss starts at 0.5 and increases gradually from the first epochs. What's the problem?

**Comment by glenn-jocher** [comment]
@sonyoungrack hello! Thank you for reaching out with your question. It's great to hear that you're using YOLOv8 for your project.

When you observe an increase in loss during continued training, it could be due to several factors. One common reason might be the learning rate. If the learning rate is not adjusted properly when resuming training, it can cause the model to diverge. Also, ensure that the optimizer state is correctly loaded along with the weights.

Another aspect to consider is the randomness in data augmentation. If the augmentations are too aggressive or varied between training sessions, this might affect the loss.

Lastly, double-check that the dataset is being loaded consistently between sessions and that there are no changes in the preprocessing or data pipeline.

For a more detailed investigation, you might want to review the training hyperparameters and compare the environment settings between the two sessions.

For further guidance, you can refer to our documentation on the Train mode at https://docs.ultralytics.com/modes/train, which might provide additional insights into the training process.

Keep experimenting, and don't hesitate to reach out if you have more questions! 🚀

**Comment by sonyoungrack** [comment]
@glenn-jocher 
my code is this.
`
model = YOLO('/content/drive/MyDrive/YOLO_models/' + use_version + '/train/weights/last.pt')
model.train(
    data='/content/datasets/data.yaml',
    patience=200,
    exist_ok=True,
    name='train',
    save=True,
    project=save_path,
    batch=-1,
    epochs=200)
`

What I want is to inherit the weight of the loaded model as it is and proceed with the learning. The last cls_loss of the loaded model was 0.319,

The dataset and hyperparameters are the same as before without any changes

The current progress is 0.4 on the first epochs as shown below,
There is an increase in cls_loss than before, and even as epochs progresses, the loss increases.

Am I doing something wrong?


`Ultralytics YOLOv8.1.5 🚀 Python-3.10.12 torch-2.1.0+cu121 CUDA:0 (NVIDIA A100-SXM4-40GB, 40514MiB)
engine/trainer: task=detect, mode=train, model=/content/drive/MyDrive/YOLO_models/240124_02/train/weights/last.pt, data=/content/datasets/data.yaml, epochs=200, time=None, patience=200, batch=200, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=/content/drive/MyDrive/YOLO_models/240124_03, name=train, exist_ok=True, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=/content/drive/MyDrive/YOLO_models/240124_03/train

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 22        [15, 18, 21]  1    758722  ultralytics.nn.modules.head.Detect           [38, [64, 128, 256]]          
Model summary: 225 layers, 3018258 parameters, 3018242 gradients, 8.2 GFLOPs

Transferred 355/355 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir /content/drive/MyDrive/YOLO_models/240124_03/train', view at http://localhost:6006/
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
train: Scanning /content/datasets/train/labels.cache... 25758 images, 10260 backgrounds, 0 corrupt: 100%|██████████| 25758/25758 [00:00<?, ?it/s]
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
val: Scanning /content/mahjong_new/valid/labels.cache... 1317 images, 412 backgrounds, 0 corrupt: 100%|██████████| 1317/1317 [00:00<?, ?it/s]
Plotting labels to /content/drive/MyDrive/YOLO_models/240124_03/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0015625), 63 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /content/drive/MyDrive/YOLO_models/240124_03/train
Starting training for 200 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/200      28.2G      0.651     0.4273     0.8896        684        640: 100%|██████████| 129/129 [01:32<00:00,  1.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:05<00:00,  1.32s/it]                   all       1317       3128      0.997      0.994      0.995       0.92


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/200      28.5G     0.6904     0.4584     0.8999        677        640: 100%|██████████| 129/129 [01:29<00:00,  1.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:04<00:00,  1.12s/it]                   all       1317       3128      0.996      0.996      0.995      0.893


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/200        28G     0.7628     0.5249     0.9269        615        640: 100%|██████████| 129/129 [01:30<00:00,  1.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:04<00:00,  1.19s/it]                   all       1317       3128      0.994      0.985      0.994      0.845


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      4/200      27.6G     0.8283     0.5805     0.9519        681        640: 100%|██████████| 129/129 [01:29<00:00,  1.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:04<00:00,  1.24s/it]                   all       1317       3128      0.998       0.98      0.994      0.856


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      5/200      27.6G     0.8428     0.5943     0.9571        689        640: 100%|██████████| 129/129 [01:28<00:00,  1.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:04<00:00,  1.18s/it]                   all       1317       3128      0.987      0.984      0.994      0.854


      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      6/200      27.1G     0.8501     0.6027      0.966        639        640: 100%|██████████| 129/129 [01:30<00:00,  1.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:04<00:00,  1.16s/it]                   all       1317       3128      0.983      0.984      0.993      0.834`

**Comment by glenn-jocher** [comment]
@sonyoungrack, it appears that your training setup is correct, and you're loading the weights as intended. The increase in cls_loss could be due to various reasons, such as slight differences in the training environment or the stochastic nature of the training process. It's also possible that the model is experiencing some overfitting.

Make sure that the optimizer state is loaded correctly along with the model weights. Also, consider using a learning rate scheduler to adjust the learning rate during training, as this can sometimes help in stabilizing the training process.

If the issue persists, you might want to experiment with different hyperparameters or even review your dataset for any inconsistencies. Keep an eye on the validation loss as well to ensure that the model is generalizing well.

For more detailed guidance on training, you can refer to our documentation on the Train mode. Keep up the good work, and don't hesitate to ask if you have more questions! 🌟

**Comment by Teassassin** [comment]
I've encountered the exactly same issue. The loss in 2nd session was just not consistent with the 1st session until 'resume=True' was set.
It seems that 'resume=True' makes optimizer state loaded correctly, solving the problem of inconsistent loss. But setting 'resume=True' will stop me from using new hyper params to finetune the model.
Is there any other ways to load optimizer state without using 'resume=True'?

**Comment by glenn-jocher** [comment]
@Teassassin hey there! 😊

Great observation! When you notice a difference in loss across sessions and setting `resume=True` helps, it indeed points towards the optimizer state being involved in maintaining continuity in training.

While `resume=True` directly brings in the optimizer state, if you wish to tweak hyperparameters without using `resume=True`, you might consider manually adjusting the optimizer's parameters right after loading them. Here's a quick example:

```python
model = YOLO('path/to/last.pt', resume=True)  # This loads model and optimizer state
model.hyp['lr0'] = 0.001  # Adjust new learning rate
model.hyp['momentum'] = 0.9  # Adjust new momentum
# Proceed to train with new hyperparameters
```

This way, you load the previous state and then manually adjust hyperparameters as needed. Hope this helps, and happy finetuning! 🚀
